zrml/swaps/src/weights.rs

```diff
@@ -80,45 +80,51 @@ impl<T: frame_system::Config> WeightInfoZeitgeist for WeightInfo<T> {
+        (57_698_000 as Weight)
+            // Standard Error: 2_000
+            .saturating_add((1_027_000 as Weight).saturating_mul(a as Weight))
             .saturating_add(T::DbWeight::get().reads(3 as Weight))
             .saturating_add(T::DbWeight::get().writes(1 as Weight))
     }
     // Storage: MarketCommons Markets (r:1 w:0)
     // Storage: MarketCommons MarketPool (r:1 w:0)
     // Storage: Swaps Pools (r:1 w:1)
     fn admin_clean_up_pool_cpmm_scalar() -> Weight {
-        (47_801_000 as Weight)
+        (52_361_000 as Weight)
             .saturating_add(T::DbWeight::get().reads(3 as Weight))
             .saturating_add(T::DbWeight::get().writes(1 as Weight))
     }
-    // Storage: Swaps Pools (r:1 w:1)
-    // Storage: Swaps SubsidyProviders (r:11 w:10)
-    // Storage: Tokens Accounts (r:22 w:22)
-    // Storage: System Account (r:11 w:11)
-    // Storage: Tokens TotalIssuance (r:2 w:2)
-    // Storage: RikiddoSigmoidFeeMarketEma RikiddoPerPool (r:1 w:0)
-    fn end_subsidy_phase(a: u32, b: u32) -> Weight {
+    // Storage: Swaps PoolsCachedForArbitrage (r:7 w:6)
+    // Storage: Swaps Pools (r:6 w:0)
+    // Storage: Tokens Accounts (r:396 w:396)
+    // Storage: System Account (r:6 w:0)
+    // Storage: Tokens TotalIssuance (r:64 w:64)
+    fn apply_to_cached_pools_execute_arbitrage(a: u32) -> Weight {
         (0 as Weight)
-            // Standard Error: 186_000
-            .saturating_add((31_081_000 as Weight).saturating_mul(a as Weight))
-            // Standard Error: 1_184_000
-            .saturating_add((143_417_000 as Weight).saturating_mul(b as Weight))
-            .saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(a as Weight)))
-            .saturating_add(T::DbWeight::get().reads((9 as Weight).saturating_mul(b as Weight)))
-            .saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(a as Weight)))
-            .saturating_add(T::DbWeight::get().writes((9 as Weight).saturating_mul(b as Weight)))
+            // Standard Error: 905_000
+            .saturating_add((3_336_767_000 as Weight).saturating_mul(a as Weight))
+            .saturating_add(T::DbWeight::get().reads(44 as Weight))
+            .saturating_add(T::DbWeight::get().reads((69 as Weight).saturating_mul(a as Weight)))
+            .saturating_add(T::DbWeight::get().writes(44 as Weight))
+            .saturating_add(T::DbWeight::get().writes((67 as Weight).saturating_mul(a as Weight)))
+    }
+    // Storage: Swaps PoolsCachedForArbitrage (r:7 w:6)
+    fn apply_to_cached_pools_noop(a: u32) -> Weight {
+        (0 as Weight)
+            // Standard Error: 7_000
+            .saturating_add((10_520_000 as Weight).saturating_mul(a as Weight))
+            .saturating_add(T::DbWeight::get().reads(1 as Weight))
+            .saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(a as Weight)))
+            .saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(a as Weight)))
```

Comment:
Hopefully the arbitrage doesn't stall our chain in the end. 🤞 I know the weight problem is known and I already had a couple of comments in the review about this, but just that we keep this in our minds if the chain stalls for unknown reasons.